installer/src/main/bin/shared.cmd

```diff
@@ -41,11 +41,48 @@ GOTO :ENDFUNCTIONS
       EXIT /B 2
     )
 
-    FOR /F %%i IN ('%JAVA_EXE% -version 2^>^&1') DO (
-      IF "%%i" == "OpenJDK" (
-        ECHO JAVA_HOME %JAVA_HOME% contains OpenJDK^, which is not supported >&2
+    SET OPEN_JDK=false
+	SET ORACLE_ONE=0
+	SET ORACLE_TWO=0
+    FOR /F "tokens=1,5" %%x IN ('%JAVA_EXE% -version 2^>^&1') DO (
+        IF "%%x" == "OpenJDK" (
+            SET OPEN_JDK=true
+	        IF EXIST %ORACLE_HOME%\wlserver\server\lib\weblogic.jar (
+			    FOR /F "tokens=1-3 delims= " %%A IN ('%JAVA_EXE% -cp %ORACLE_HOME%\wlserver\server\lib\weblogic.jar weblogic.version 2^>^&1') DO (
+					IF "%%A" == "WebLogic" (
+			            FOR /F "tokens=1-5 delims=." %%j IN ('ECHO %%C') DO (
+			                SET "ORACLE_VERSION=%%j.%%k.%%l.%%m.%%n"
+			                SET "ORACLE_ONE=%%j"
+			                SET "ORACLE_TWO=%%l"
+				        )
+			        )
+			    )
+			 	SET GRAALVM=false
+			    IF "%%y" == "GraalVM" SET GRAALVM=true
+		    ) ELSE (
+                  ECHO JAVA_HOME %JAVA_HOME% contains OpenJDK^, which is not supported >&2
+                  EXIT /B 2
+            )
+        )
+    )
+
+    SET NOT_VALID=false
+    IF "%OPEN_JDK%"=="true" (
+	    IF "%GRAALVM%"=="false" SET NOT_VALID=true
+	    IF %ORACLE_ONE% LSS 14 (
+		    SET NOT_VALID=true
+		) ELSE IF %ORACLE_ONE% EQU 14 IF %ORACLE_TWO% LSS 2 SET NOT_VALID=true
+	    SET JAVA_VENDOR=GraalVM
+	)
+
+    IF "%NOT_VALID%"=="true" (
+		IF "%GRAALVM%"=="true" (
+		   SET ORACLE_VERSION
+           ECHO JAVA_HOME %JAVA_HOME% contains GraalVM OpenJDK^, which is not supported in versions before 14.1.2 >&2
+	       EXIT /B 2
+		)
+	    ECHO JAVA_HOME %JAVA_HOME% contains OpenJDK^, which is not supported. >&2
```

Comment:
The indention is off here, looks like there is a mix of tabs and spaces on some of the new lines. looks like the rest of the cmd scripts use spaces only.